buildSrc/src/main/kotlin/it/unibo/collektive/codegen/FieldPrimitivesGenerator.kt

```diff
@@ -0,0 +1,68 @@
+package it.unibo.collektive.codegen
+
+import com.squareup.kotlinpoet.FileSpec
+import it.unibo.collektive.field.Field
+import kotlin.reflect.KCallable
+import kotlin.reflect.KClass
+import kotlin.reflect.KVisibility
+
+/**
+ * The base types for which to generate field functions.
+ */
+val baseTargetTypes = sequenceOf(
+    Boolean::class,
+    Byte::class,
+    Short::class,
+    Int::class,
+    Float::class,
+    Double::class,
+    Long::class,
+    UByte::class,
+    UShort::class,
+    UInt::class,
+    ULong::class,
+    Char::class,
+    CharSequence::class,
+    String::class,
+    Map::class,
+    List::class,
+    Collection::class,
+    Set::class,
+    Array::class,
+    ByteArray::class,
+    ShortArray::class,
+    IntArray::class,
+    FloatArray::class,
+    DoubleArray::class,
+    LongArray::class,
+    // Comparator::class,
```

Comment:
Why is Comparator commented out?

Reply:
The Kotlin `Comparator` type is an alias for the java `Comparator`. Since we are in KMP, we cannot share Java-dependent code. For this reason, I've temporarily dropped the `Comparator` from the generation.

Perhaps, in the future we must consider to generate also the jvm-specific code for the fields

Reply:
There's no `Comparator` in Kotlin-MP?